app/models/annotation.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# Annotation represents a user-supplied comment or note attached to any annotatable resource.
+#
+# == Schema Information
+#
+# Table name: annotations
+#
+#  id                 :bigint           not null, primary key
+#  annotation_type_id :bigint           not null, foreign key
+#  annotatable_type   :string           not null
+#  annotatable_id     :bigint           not null
+#  comment            :string(500)      not null
+#  user               :string(10)       not null
+#  created_at         :datetime         not null
+#  updated_at         :datetime         not null
+#
+# == Associations
+#
+# * +annotation_type+ - The type/category of the annotation.
+# * +annotatable+     - The resource (polymorphic) this annotation is attached to.
+#
+# == Validations
+#
+# * +comment+ - Must be present and at most 500 characters.
+# * +user+    - Must be present and at most 10 characters.
+#
+class Annotation < ApplicationRecord
+  belongs_to :annotation_type
+  belongs_to :annotatable, polymorphic: true
+
+  validates :comment, presence: true, length: { maximum: 500 }
+  validates :user, presence: true, length: { maximum: 10 }
```

Comment:
Just a minor thing - Do we need to validate the presence of annotable as well to not to allow orphaned annotations? - or as the runs and wells contain - dependent: :destroy, this is not going to happen?

Reply:
I don't know. It might need changes when we do the front end bit. I am happy to leave as is for now and get this merged in. I will deploy from branch to ensure it doesn't break runs.